gcloud-java-resourcemanager/src/main/java/com/google/gcloud/spi/ResourceManagerRpc.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.spi;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.services.cloudresourcemanager.model.Policy;
+import com.google.api.services.cloudresourcemanager.model.Project;
+import com.google.common.collect.ImmutableList;
+import com.google.gcloud.resourcemanager.ResourceManagerException;
+
+import java.util.Collections;
+import java.util.List;
+
+public interface ResourceManagerRpc {
+
+  public enum Permission {
+    CREATE,
+    DELETE,
+    GET,
+    LIST,
+    UPDATE,
+    GET_IAM_POLICY,
+    SET_IAM_POLICY
+  }
+
+  public enum DeleteResult {
+    SUCCESS,
+    ALREADY_DELETED
+  }
+
+  public enum UndeleteResult {
+    SUCCESS,
+    DELETE_IN_PROGRESS,
+    GONE
```

Comment:
You can only perform undelete operations on projects in the DELETE_REQUESTED state. If the project is any other state, we return a 400 bad request.

Once the project is completely expunged, we will return 403 forbidden. We do not return not found because we do not want to leak information about the existence of projects.